core/src/display_object/movie_clip.rs

```diff
@@ -1850,8 +1861,17 @@ impl<'gc> MovieClip<'gc> {
         let data = mc.static_data.swf.clone();
         let mut index = 0;
 
-        // Sanity; let's make sure we don't seek way too far.
-        let clamped_frame = frame.min(max(mc.frames_loaded(), 0) as FrameNumber);
+        let clamped_frame = if self.movie().is_action_script_3() {
+            // Make sure we don't allow a negative seek. Note that we do
+            // *not* check this against the frame count in the SWF header - flash player
+            // ignores that value and allows you to seek to any frame, even if the total
+            // number of frames exceeds the frame count in the header.
+            frame.min(max(mc.frames_loaded_ignore_header(), 0) as FrameNumber)
+        } else {
+            // Sanity; let's make sure we don't seek way too far.
+            frame.min(max(mc.frames_loaded(), 0) as FrameNumber)
```

Comment:
Does this mean gotos in AVM1 that go past the frames in the header are clamped? I made a small test and AVM1 also seems to ignore the header frame count in FP.